src/main/scala/mesosphere/marathon/MarathonModule.scala

```diff
@@ -67,6 +69,32 @@ class MarathonModule(conf: MarathonConf, http: HttpConf, zk: ZooKeeperClient)
   }
 
   @Provides
+  @Inject
+  @Singleton
+  def provideSchedulerCallbacks(schedulerService: MarathonSchedulerService): SchedulerCallbacks = {
+    new SchedulerCallbacksServiceAdapter(schedulerService)
+  }
+
+  /**
+    * Provider for [[SchedulerDriver]]s. It is no singleton by intention.
+    * As a side effect, it sets the corresponding driver in MarathonSchedulerDriver.
+    */
+  @Provides
+  @Inject
+  def provideSchedulerDriver(holder: MarathonSchedulerDriverHolder,
+                             config: MarathonConf,
+                             httpConfig: HttpConf,
+                             newScheduler: MarathonScheduler,
+                             frameworkIdUtil: FrameworkIdUtil): SchedulerDriver = {
+    import mesosphere.util.ThreadPoolContext.context
+    implicit val zkTimeout = config.zkFutureTimeout
+    val frameworkId = frameworkIdUtil.fetch
+    val driver = MarathonSchedulerDriver.newDriver(config, httpConfig, newScheduler, frameworkId)
+    holder.driver = Some(driver)
```

Comment:
I don't like the fact, that we have such functionality in the wiring part and use Provider[MarathonSchedulerDriver] elsewhere.